tests/azure_iot_provisioning_client_ut.c

```diff
@@ -0,0 +1,704 @@
+    xPacketInfo.ucType = 0;
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 ),
+                      eAzureIoTProvisioningPending );
+
+    /* Registration response */
+    prvGenerateGoodResponse( &xPublishInfo, 1 );
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 ),
+                      eAzureIoTProvisioningPending );
+
+    /* Read response */
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 ),
+                      eAzureIoTProvisioningPending );
+
+    /* Wait for timeout */
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    ullUnixTime += 2;
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 ),
+                      eAzureIoTProvisioningPending );
+
+    /* Publish Registration Query */
+    will_return( AzureIoTMQTT_Publish, eAzureIoTMQTTSuccess );
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xPacketInfo.ucType = 0;
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 ),
+                      eAzureIoTProvisioningPending );
+
+    /* Restration response */
+    prvGenerateGoodResponse( &xPublishInfo, 0 );
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 ),
+                      eAzureIoTProvisioningPending );
+
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xPacketInfo.ucType = 0;
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 ),
+                      eAzureIoTProvisioningSuccess );
+
+    AzureIoTProvisioningClient_Deinit( &xTestProvisioningClient );
+}
+/*-----------------------------------------------------------*/
+
+uint32_t ulGetAllTests()
+{
+    const struct CMUnitTest tests[] =
+    {
+        cmocka_unit_test( testAzureIoTProvisioningClient_Init_Failure ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_Init_Success ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_Deinit_Success ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_SymmetricKeySet_Failure ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_SymmetricKeySet_Success ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_Register_ConnectFailure ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_Register_SubscribeFailure ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_Register_SubscribeAckFailure ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_Register_PublishFailure ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_Register_RegistrationFailure ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_Register_QueryFailure ),
+        cmocka_unit_test( testAzureIoTProvisioningClient_Register_Success )
```

Comment:
I'm assuming stuff like `AzureIoTProvisioningClient_GetDeviceAndHub` is being tested in a later PR?  If so that's actually great since it keeps this PR smaller but want to make sure.?

Reply:
Yes I will create new PR. Right now code coverage is 76, which is something I am not happy with. I will add additional tests to raise this code coverage.

---
In reply to: [607961133](https://github.com/Azure/azure-iot-middleware-freertos/pull/62#discussion_r607961133) [](ancestors = 607961133)